### PR TITLE
Docs: Update example of results returned from `executeOnFiles`

### PR DESCRIPTION
**Pull request by SimenB**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

I just noticed `severity` being there twice. Then I though to double check what the output actually is, and it's quite different, so I changed the whole thing.

**Please check each item to ensure your pull request is ready:**
- [x] I've read the [pull request guide](http://eslint.org/docs/developer-guide/contributing/pull-requests)
- [ ] I've included tests for my change
- [ ] I've updated documentation for my change (if appropriate)

**What changes did you make? (Give an overview)**
Replace example of returned object from `executeOnFiles`.

**Is there anything you'd like reviewers to focus on?**
- Should other examples be updated as well?
- `fatal: false` is missing, should it? I suppose formatters just do `if (false)` or something like that, so if it's `undefined` or `false` doesn't really matter

### CODE REVIEW

File: docs/developer-guide/nodejs-api.md [hunk 1]
```diff
@@ -220,21 +220,20 @@ The return value is an object containing the results of the linting operation. H
...
-                    ruleId: "semi",
-                    severity: 2,
-                    line: 1,
-                    column: 23,
-                    message: "Expected a semicolon."
-                }
-            ],
+            filePath: "/Users/eslint/project/myfile.js",
+            messages: [{
+                ruleId: "semi",
```
Comment (SimenB) [inline docs/developer-guide/nodejs-api.md]: This is just `JSON.stringify` with removed quotes from keys. Should the order match the old order?

File: docs/developer-guide/nodejs-api.md [hunk 2]
```diff
@@ -220,21 +220,20 @@ The return value is an object containing the results of the linting operation. H
...
+                severity: 2,
+                message: "Missing semicolon.",
+                line: 1,
+                column: 13,
+                nodeType: "ExpressionStatement",
+                source: "\"use strict\"",
```
Comment (vitorbal) [inline docs/developer-guide/nodejs-api.md]: You could add an inline comment pointing to the deprecation notice on line 254:
```js
// Deprecated: see "please note" paragraph below.
```

**Comment by vitorbal** [comment]
Thanks, @SimenB! I think it's fine to omit `fatal`.
You could certainly open other PR's to fix more of these examples if you find more problems, would be greatly appreciated!

LGTM, waiting for others to take a look.

**Comment by platinumazure** [comment]
So it looks like you've replaced a fixable example (semi) with a non-fixable example (strict). Is there any way we could have one file with each case, so that we could see the `output` property present in only one of the file outputs? Or is there a reason for switching the example to non-fixable that I'm not aware of?

**Comment by SimenB** [comment]
@platinumazure The rule id giving an error is still semi 😄

**Comment by platinumazure** [comment]
Oops, I wasn't reading carefully enough.

I definitely don't see an "output" property, though. Was that definitely not present in your results?

**Comment by SimenB** [comment]
Good point, didn't even notice!

Shorter sample:

```js
const { CLIEngine } = require('./');
const instance = new CLIEngine({ rules: { semi: 'error' } });
const res = instance.executeOnText('"use strict"\n', 'filename.js')

console.log(JSON.stringify(res, null, 2));
```

Gives:

```js
{
  "results": [
    {
      "filePath": "/Users/simen/Development/eslint/filename.js",
      "messages": [
        {
          "ruleId": "semi",
          "severity": 2,
          "message": "Missing semicolon.",
          "line": 1,
          "column": 13,
          "nodeType": "ExpressionStatement",
          "source": "\"use strict\"",
          "fix": {
            "range": [
              12,
              12
            ],
            "text": ";"
          }
        }
      ],
      "errorCount": 1,
      "warningCount": 0,
      "source": "\"use strict\"\n"
    }
  ],
  "errorCount": 1,
  "warningCount": 0
}
```

**Comment by SimenB** [comment]
Seems like the fix itself is included, but not the source after applying it (`output`)

**Comment by vitorbal** [comment]
That's how it looks like when there are fixes but they were not applied, no?

**Comment by SimenB** [comment]
Then the docs here are way wrong, especially http://eslint.org/docs/developer-guide/nodejs-api#outputfixes

**Comment by platinumazure** [comment]
Oh, I just noticed you aren't using `fix: true` in your CLIEngine constructor options argument. So we include the raw fix information but don't bother computing an output.

**Comment by SimenB** [comment]
Oh, didn't know about that option.

The bottom example is still wrong, then (copied here):

```js
var CLIEngine = require("eslint").CLIEngine;

var cli = new CLIEngine({
    envs: ["browser", "mocha"],
    useEslintrc: false,
    rules: {
        semi: 2
    }
});

// lint myfile.js and all files in lib/
var report = cli.executeOnFiles(["myfile.js", "lib/"]);

// output fixes to disk
CLIEngine.outputFixes(report);
```

**Comment by vitorbal** [comment]
@SimenB sorry, I'm not sure what you mean by that. Could you explain why you think that example is wrong?

**Comment by SimenB** [comment]
As it says that the thing returned from the execution of `cli.executeOnFiles(["myfile.js", "lib/"]);` can output fixes. It cant, as `cli` was not instantiated with `fix: true`

**Comment by platinumazure** [comment]
@SimenB Okay, I agree that last example is wrong and needs to be changed. We can change that in a separate PR, unless you want to add that to this PR (I think the only necessary change is adding `fix: true` to the CLI options).

For now, here's what I propose:

1. Show an example of what output would look like without `fix: true` (or `--fix`), meaning show the `fix` blocks in messages and don't show output anywhere. (This is what you have more or less)
1. Show an example of what output looks like with `fix: true`, and in that output, show one "file" that has one or more fixes that can be applied (i.e., has `output`) and one that has no fixes (i.e., does not have `output`).
1. Optionally, fix the issue you just brought up as well (if not, we can do that in a separate PR).

If we can get those first two examples documented well, I think that will be a huge benefit to this area of the documentation.

What do you think @SimenB?

**Comment by SimenB** [comment]
@platinumazure Done!

One thing I noticed is that in `fix` mode, `source` is omitted from the file (the thing introduced in 3.7). Is that on purpose? Seems like a bug to me.

And thinking about the `fatal:  false`, it's very useful for documentation at least if it's present, if people are creating custom formatters. I'll send a separate PR adding it.

**Comment by vitorbal** [comment]
@SimenB it's indeed on purpose, as it would be redundant when `output` is present. You can read the comments on the PR you mention for more details: https://github.com/eslint/eslint/pull/7304#discussion_r81833872

**Comment by SimenB** [comment]
So every formatter must do `output || source`? Why not just have the same string in both places?

**Comment by nzakas** [comment]
Just a heads up: we have moved to a new CLA checker on pull requests. Even if you've previously signed our CLA, we will need to you sign the new one. To do so, look at the status checks for **licence/cla** and click the "Details" link. Sorry for the inconvenience.

**Comment by SimenB** [comment]
I completely forgot about this. Rebased (and signed CLA) now

**Pull request merged**